### PR TITLE
feat: enable custom configurations for LimeSurvey service

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
### Description
This PR adds a configuration file during the image-building process so it can be used by the [LimeSurvey docker entrypoint](https://github.com/adamzammit/limesurvey-docker/blob/master/docker-entrypoint.sh#L58) when starting the service to populate its configurations. The base file config.php was copied from a remote installation and then modified by appending:

```
         'request' => array(
			'enableCsrfValidation'=>true,
			'csrfCookie' => array(
				'sameSite' => 'None',
				'secure' => true,
			),
		),
```
and
```
 'x_frame_options' => 'allow',
```
So the LimeSurvey installations allow being embedded in domains different than theirs.

This PR solves this customization issue: https://github.com/eduNEXT/tutor-contrib-limesurvey/issues/6

### How to test
1. Install this `tutor-contrib-limesurvey` version
2. Run `tutor config save`
3. Run `tutor dev start -d`
4. Create a new LimeSurvey component, click `Edit` and add our remote installation under LimeSurvey URL:
image
You should be able to perform any operation when visiting the LimeSurvey tab from the instructor dashboard.

### CODE REVIEW

**Comment by BryanttV** [comment]
I'm currently getting a CRSF error at my local:
image
Also, I can't log in to our remote installation, when I enter the credentials and press the "login" button it just reloads the page.

File: tutorlimesurvey/plugin.py [hunk 1]
```diff
@@ -20,15 +20,16 @@
         # Each new setting is a pair: (setting_name, default_value).
         # Prefix your setting names with 'LIMESURVEY_'.
         ("LIMESURVEY_VERSION", __version__),
-        ("LIMESURVEY_DOCKER_IMAGE", "docker.io/acspri/limesurvey:6.1.0"),
+        ("LIMESURVEY_DOCKER_IMAGE", "docker.io/edunext/limesurvey:15.0.0"),
+        ("LIMESURVEY_DOCKER_BASE_IMAGE", "docker.io/acspri/limesurvey:6.1.0"),
```
Comment (BryanttV) [inline tutorlimesurvey/plugin.py:24]: Should we start using [this image](https://hub.docker.com/r/adamzammit/limesurvey)?

**Comment by mariajgrimaldi** [comment]
@BryanttV addressing your comments:

> I'm currently getting a CRSF error at my local: image 

I'm not sure what's going on here, I'll check that out :)

> Also, I can't log in to our remote installation, when I enter the credentials and press the "login" button it just reloads the page.
It's working for me even in a private window, not sure if it's something related to your browser. Can you try removing cookies and stuff?

**Comment by mariajgrimaldi** [comment]
These are the test cases I'm currently considering @BryanttV:
1. Login to LimeSurvey service outside the iframe
2. Login to LimeSurvey service inside from the instructor dashboard iframe
In local and remote environments, both of them using the `docker.io/edunext/limesurvey:15.0.0` with the current `config.php` configurations.

Can you specify where it's not working for you? With all the needed details: environment and test case, be as specific as you can so I can address those issues. Thanks!

**Comment by mariajgrimaldi** [comment]
I have a theory about this working in dev environments:

**Comment by BryanttV** [comment]
According to my tests I obtained the following:

#### local environment
1. **Outside the iframe**: I'm getting a CSRF token error when I log in.
2. **Inside the instructor dashboard iframe:** I'm getting the same CSRF token error when I log in.

#### remote environment
1. **Outside the iframe:** I can log in without any problem.
2. **Inside the instructor dashboard iframe:** I can't log in. When I enter the credentials and press the "login" button it just reloads the page.

**Comment by mariajgrimaldi** [comment]
@BryanttV can you test the last changes I uploaded for the local environment? We're now using the same image as before for dev environments. Check out why [here](https://github.com/eduNEXT/tutor-contrib-limesurvey/pull/9#issuecomment-1664187587).

Now, regarding the error in the instructor dashboard iframe, did you try testing it in a private window?

**Comment by BryanttV** [comment]
Perfect!
1. Now it works in the local environment, both in the iframe of the instructor's panel and outside of it.
2. It's also working in the remote environment, I just had to use a different browser (firefox)

File: tutorlimesurvey/plugin.py (hunk 1 above)
Comment (BryanttV) [reply tutorlimesurvey/plugin.py:24]: You made reference to [this repository](https://github.com/adamzammit/limesurvey-docker) in the description of the PR.

Comment (Ian2012) [reply tutorlimesurvey/plugin.py:24]: This variable is not used anywhere, does it have any pupose?

File: tutorlimesurvey/plugin.py [hunk 2]
```diff
@@ -20,15 +20,17 @@
         # Each new setting is a pair: (setting_name, default_value).
         # Prefix your setting names with 'LIMESURVEY_'.
         ("LIMESURVEY_VERSION", __version__),
-        ("LIMESURVEY_DOCKER_IMAGE", "docker.io/acspri/limesurvey:6.1.0"),
+        ("LIMESURVEY_DOCKER_IMAGE", "docker.io/edunext/limesurvey:15.0.0-tmp.0.1"),
```
Comment (Ian2012) [inline tutorlimesurvey/plugin.py:23]: Can you use `DOCKER_IMAGE_LIMESURVEY` ?

File: tutorlimesurvey/templates/limesurvey/build/limesurvey/config.php
Comment (Ian2012) [inline tutorlimesurvey/templates/limesurvey/build/limesurvey/config.php:1]: Can you leave a source for this file so in the future maintainers will now how to verify this file?

File: tutorlimesurvey/plugin.py (hunk 2 above)
Comment (mariajgrimaldi) [reply tutorlimesurvey/plugin.py:23]: I thought about it but the in-line comments (from the cookie cutter) say `# Prefix your setting names with 'LIMESURVEY_'`

File: tutorlimesurvey/templates/limesurvey/build/limesurvey/config.php
Comment (mariajgrimaldi) [reply tutorlimesurvey/templates/limesurvey/build/limesurvey/config.php:1]: That's a good idea! Sure do

**Comment by BryanttV** [comment]
LGTM

**Review by BryanttV — APPROVED** [review]

File: tutorlimesurvey/templates/limesurvey/build/limesurvey/config.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            'emulatePrepare' => true,
+            'username' =>'{{ LIMESURVEY_DB_USER }}',
+            'password' =>'{{ LIMESURVEY_DB_PASSWORD }}',
+            'charset' => 'utf8mb4',
+'attributes' => array(),
+            'tablePrefix' =>'',
+        ),
+        {%- raw %}
+        'session' => array (
+			'sessionName'=>'LS-VBXZZORFJZCAKGYI',
```
Comment (felipemontoya) [inline tutorlimesurvey/templates/limesurvey/build/limesurvey/config.php:56]: Is it reasonable to leave a fixed session name?

Is this something that should be auto generated?

File: tutorlimesurvey/plugin.py (hunk 1 above)
Comment (felipemontoya) [reply tutorlimesurvey/plugin.py:24]: I suppose that is what we use to build our own, right?

Also, the image itself makes no difference. It is published by the same person in both places.

but we should move to 6.2.0 which is now available. https://hub.docker.com/layers/acspri/limesurvey/6.2.0/images/sha256-7020d0a3e849726efe400ace3d27e2fc270d77bb7cb914b8f2e8b97b8edf44b6?context=explore

Comment (mariajgrimaldi) [reply tutorlimesurvey/plugin.py]: Yes, we built the production image on top of it. Thanks!

File: tutorlimesurvey/templates/limesurvey/build/limesurvey/config.php (hunk 3 above)
Comment (mariajgrimaldi) [reply tutorlimesurvey/templates/limesurvey/build/limesurvey/config.php:56]: I copied this from the docs: https://manual.limesurvey.org/Optional_settings#Allow_usage_of_session_and_Csrf_Validation_in_iFrame. It's something I prefer to leave as is.

**Review by Ian2012 — APPROVED** [review]

**Pull request merged**